packages/server/src/api/rest/index.ts

```diff
@@ -405,7 +408,12 @@ class RequestHandler extends APIHandlerBase {
             include = allIncludes;
         }
 
-        const entity = await prisma[type].findUnique(args);
+        let entity = await prisma[type].findUnique(args);
+
+        if (typeInfo.idFields.length > 1) {
+            entity = { ...entity, [compoundIdKey]: resourceId };
```

Comment:
It seems the compound id field is only included when reading a single entity. Should it be included when reading a collection or via a relation too?

Reply:
Yes it should always be included I think. Will look at that when adding tests for those situations.